src/main/java/com/fasterxml/jackson/core/io/LazyBigDecimal.java

```diff
@@ -0,0 +1,40 @@
+package com.fasterxml.jackson.core.io;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * This implementation stores the {@link BigDecimal} as text until it is needed. It can be expensive
+ * to parse a number and in some cases, the number will never be retrieved.
+ *
+ * @since 2.15
+ */
+public class LazyBigDecimal implements LazyNumber {
+    private final String _value;
+    private final boolean _useFastParser;
+    private BigDecimal _decimal;
+
+    public LazyBigDecimal(final String value, final boolean useFastParser) {
```

Comment:
Makes sense, although one additional consideration is that of binary formats: in those cases construction is with known `BigDecimal` but missing `String _value`. Same for other LazyXxx implementations.
Would it be easy enough to change things to allow construction either way?
With that I think we could get support for `TokenBuffer` quite easily done -- I like the way this is going.

Oh, one other possible complication: `Double` vs `BigDecimal` -- conversion is potentially lossy and ineffecient. Maybe we just need `LazyDouble` (and I suppose `LazyFloat`).

Reply:
I have constructors on the LazyBig* classes that take existing Big* instances and that don't need the string value.

The Double/Floats are lossy so LazyBig* classes are needed but I can add LazyDouble/LazyFloat too.